src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -2968,15 +3003,41 @@ void AutomationInstrumentClipView::handleSinglePadPress(ModelStackWithTimelineCo
 				effectiveLength = clip->loopLength;
 			}
 
-			uint32_t squareStart = getPosFromSquare(xDisplay);
-			uint32_t squareWidth = instrumentClipView.getSquareWidth(xDisplay, effectiveLength);
-			if (squareWidth != 3) {
-				squareStart = squareStart + (squareWidth / 2);
+			uint32_t squareStart = 0;
+
+			//if a long press is selected and you're checking value of start or end pad
+			//display value at very first or very last node
+			if (multiPadPressSelected && ((leftPadSelectedX == xDisplay) || (rightPadSelectedX == xDisplay))) {
+				if (leftPadSelectedX == xDisplay) {
+					squareStart = getPosFromSquare(xDisplay);
+				}
+				else {
+					int32_t squareRightEdge = getPosFromSquare(rightPadSelectedX + 1);
+					squareStart = std::min(effectiveLength, squareRightEdge) - kParamNodeWidth;
+				}
+			}
+			//display pad's middle value
+			else {
+				squareStart = getPosFromSquare(xDisplay);
+				uint32_t squareWidth = instrumentClipView.getSquareWidth(xDisplay, effectiveLength);
+				if (squareWidth != 3) {
+					squareStart = squareStart + (squareWidth / 2);
+				}
 			}
 
 			int32_t knobPos = getParameterKnobPos(modelStackWithParam, squareStart);
 			displayParameterValue(knobPos + kKnobPosOffset);
 
```

Comment:
for the pad selected in pad selection mode, set the mod position to the position corresponding to the value that was displayed. this is needed so that the selected pad's value can be auditioned